types/map.go

```diff
@@ -254,7 +254,7 @@ func (m Map) String() string {
 		if i != 0 {
 			res.WriteString(",")
 		}
-		res.WriteString(fmt.Sprintf(`"%s":%s`, k, m.Elems[k].String()))
+		res.WriteString(fmt.Sprintf("%q:%s", k, m.Elems[k].String()))
```

Comment:
We can definitely consider it! I think I slightly lean towards not doing it because:

- Object attribute names must be valid HCL identifiers so alphanumeric (not starting with a number), hyphens, and underscores
- If we're trying to slightly mimic HCL syntax with these, I don't think you can quote identifiers
- Not having the quotes may provide a slight visual clue that its an object vs map

Of course this is very subjective based on my personal opinion -- very happy to discuss this further.